### PR TITLE
LPS-191051 Remove log4j1 bridge

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang, the previous pr is https://github.com/liferay-core-infra/liferay-portal/pull/5084

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:service-builder

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
@yuhai What's this commit for? Didn't we already remove the Log4J 1 config xml support? https://github.com/liferay-core-infra/liferay-portal/pull/5097/commits/2268219dbbebd54200dcc0da9174401e4b8feb11

**Comment by yuhai** [comment]
> @yuhai What's this commit for? Didn't we already remove the Log4J 1 config xml support? [2268219](https://github.com/liferay-core-infra/liferay-portal/commit/2268219dbbebd54200dcc0da9174401e4b8feb11)

@dantewang yes, in order to keep xml log output consistent with log4j1, we used Log4j1XmlLayout from log4j-1.2-api because log4j2 didn't compatible with log4j1 xml format output. Please see [Log4j1XmlLayout](https://github.com/yuhai/liferay-portal/blob/LPS-191051-pr2/portal-impl/src/META-INF/portal-log4j.xml#L18) in portal-log4j.xml and the [Log4j1XmlLayout class](https://github.com/apache/logging-log4j2/blob/2.x/log4j-1.2-api/src/main/java/org/apache/log4j/layout/Log4j1XmlLayout.java#L38)

**Comment by dantewang** [comment]
> > @yuhai What's this commit for? Didn't we already remove the Log4J 1 config xml support? [2268219](https://github.com/liferay-core-infra/liferay-portal/commit/2268219dbbebd54200dcc0da9174401e4b8feb11)
> 
> @dantewang yes, in order to keep xml log output consistent with log4j1, we used Log4j1XmlLayout from log4j-1.2-api because log4j2 didn't compatible with log4j1 xml format output. Please see [Log4j1XmlLayout](https://github.com/yuhai/liferay-portal/blob/LPS-191051-pr2/portal-impl/src/META-INF/portal-log4j.xml#L18) in portal-log4j.xml and the [Log4j1XmlLayout class](https://github.com/apache/logging-log4j2/blob/2.x/log4j-1.2-api/src/main/java/org/apache/log4j/layout/Log4j1XmlLayout.java#L38)

I see, it's for output layout, not config.

**Comment by yuhai** [comment]
> > > @yuhai What's this commit for? Didn't we already remove the Log4J 1 config xml support? [2268219](https://github.com/liferay-core-infra/liferay-portal/commit/2268219dbbebd54200dcc0da9174401e4b8feb11)
> > 
> > 
> > @dantewang yes, in order to keep xml log output consistent with log4j1, we used Log4j1XmlLayout from log4j-1.2-api because log4j2 didn't compatible with log4j1 xml format output. Please see [Log4j1XmlLayout](https://github.com/yuhai/liferay-portal/blob/LPS-191051-pr2/portal-impl/src/META-INF/portal-log4j.xml#L18) in portal-log4j.xml and the [Log4j1XmlLayout class](https://github.com/apache/logging-log4j2/blob/2.x/log4j-1.2-api/src/main/java/org/apache/log4j/layout/Log4j1XmlLayout.java#L38)
> 
> I see, it's for output layout, not config.

@dantewang, yes

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by Kvale1733** [comment]
ci:test:sf

**Comment by Kvale1733** [comment]
@yuhai No regressions. SF issue is unrelated.

File: modules/apps/portal/portal-log4j/src/main/java/com/liferay/portal/log4j/internal/Log4j1XmlLayout.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+/**
+ * @author Hai Yu
+ * @see org.apache.log4j.layout.Log4j1XmlLayout
+ */
+@Plugin(
+	category = Node.CATEGORY, elementType = Layout.ELEMENT_TYPE,
+	name = Log4j1XmlLayout.PLUGIN_NAME, printObject = true
+)
+public final class Log4j1XmlLayout extends AbstractStringLayout {
```
Comment (dantewang) [inline modules/apps/portal/portal-log4j/src/main/java/com/liferay/portal/log4j/internal/Log4j1XmlLayout.java:42]: @yuhai Then let's just call it `XmlLayout`, no need to emphasize it's from Log4J 1.

File: modules/apps/portal/portal-log4j/src/main/java/com/liferay/portal/log4j/internal/Log4j1XmlLayout.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+	@Override
+	public void encode(
+		LogEvent logEvent, ByteBufferDestination byteBufferDestination) {
+
+		StringBuilder sb = getStringBuilder();
+
+		_formatTo(logEvent, sb);
+
+		Encoder<StringBuilder> encode = getStringBuilderEncoder();
```
Comment (dantewang) [inline modules/apps/portal/portal-log4j/src/main/java/com/liferay/portal/log4j/internal/Log4j1XmlLayout.java:59]: @yuhai Variable should be called `encoder`.

**Review by dantewang — CHANGES_REQUESTED** [review]
@yuhai Please do further formatting and clean-up, and maybe use Liferay utils to replace the Log4J utils, in the copied `XmlLayout`.

**Pull request closed without merging**